### ISSUE TITLE
ExtractArchive 

### ISSUE THREAD

**Issue by sveni-lee** [pyload/pyload#1136]
delete archives after is extracted dosen't work anymore...
there ist no error in debug log...

**Comment by Ghoststage** [comment]
Leider ist das seit anfang diesen jahres normal das nichts mehr richtig
geht die coder scheinen burnout zu haben sie bekommen nichts mehr richtig
hin

2015-02-02 14:38 GMT+01:00 sveni-lee <notifications@github.com>:

**Comment by sveni-lee** [comment]
naja, ganz so hart sollte man nicht ins Gericht gehen... Bei mir ging es ja bis zum letzten Update. Ich hate einfach die ExtractArchive.py dementsprechend auf True geändert... Aber das funktioniert jetzt auch nicht...
Nicht vergessen, die Coder inverstieren ihre Freizeit in das Projekt!

**Comment by vuolter** [comment]
I'm just testing a new version...

**Comment by zlep** [comment]
Das kann nicht sein @sveni-lee Das Entpacken funktioniert schon seit Wochen nicht mehr. Entweder hast du länger nicht geupdatet oder du hast es nicht genutzt.

**Comment by Ghoststage** [comment]
@zlep

Danke!

Am 2. Februar 2015 um 17:08 schrieb zlep <notifications@github.com>:

**Comment by sveni-lee** [comment]
@vuolter thanks...

@zlep hmm komisch, ich habe update auf 24h eingestellt und das funktioniert auch...
wie schin im Forum beschieben habe ich die delete funktion im Plugin und in der plugin.conf auf true gestellt und auch das passwort immer per Handeingegeben... Mit dem entpacken hatte ich keine Probleme nur mit dem löschen der rar datein

**Comment by immenz** [comment]
@vuolter Are you still working on ExtractArchive? I found a couple more things that don't work as expected

**Comment by sveni-lee** [comment]
With the new version I allways get the Error "ExtractArchive: MaHe0201.part1.rar | Wrong password"
The pw is right because when I extrac manuely it works

**Comment by kasperj** [comment]
Passwort file isn't working again, it was working for a short time after the rewrite

**Comment by immenz** [comment]
This RegEx works properly:

    (.+)\s+(\D+)\s+(\d+)\s+\d\d-\d\d-\d\d\s+\d\d:\d\d\s+(.+)

Maybe add some Question Marks if some Infos aren't there all the time

**Comment by immenz** [comment]
More Issues: When extraction ok, only first file of multipart Download is deleted (part 1 of 3 in total)

**Comment by zlep** [comment]
Nice work @immenz 
Did you get the Plugin working? If so, could you please upload it. I would love to take it and turn off the Updatemanager. I just want this freaking Plugin working again.
Thanks.

**Comment by sveni-lee** [comment]
@immenz 
thanks for sharing... so extracking with password-list works again for me...
But I still have a problem with deleting *.rar files. The plugin only delete the first .rar file...

**Comment by immenz** [comment]
@sveni-lee 
yes, this problem still remains
@zlep 
look in the pull request, there are the changes I made

**Comment by zlep** [comment]
Thanks.
Unfortunately deleting finished parts still not working.
And what do you mean by: "also it seems not to extract when package finished (although option Wait for all downloads to be finished: is false)"?
Thanks

**Comment by immenz** [comment]
yes, deleting just one file. that was mentioned above.

I testet my changes with one package containing 3 or 4 files. After i comitted my changes, i ran 4 packages in a row. And from log it seemed like when one package finished, it not started extracting immediate. In Fact, only 2 of 4 packages were extracted. But it was late, i decided not to look into it right then.

**Comment by zlep** [comment]
Okay, thank you. Are you trying to get these 2 issues fixed? Would be great. We would finally get a working Plugin then.

**Comment by immenz** [comment]
I will look in it this evening. 

But unfortunately fixing this 2 issues may be just do the trick for us, but there are still some other issues I simple don't discovered yet or I have no clue about it. For example SevenZip just don't work for me. 

@vuolter on the other hand always tries to take care of all Use-Cases and thats why there isn't a "simple solution" for all...

**Comment by zlep** [comment]
Yeah sure, but it would be nice if it would work properly in 95% of all cases and then fixing the rest. Right now it does not work entirely.
Thanks for your work.

**Comment by sveni-lee** [comment]
I test a little bit and for me only the first .rar file will be extracted and after this the file is deleted.
So it looks like that all extracted files will be deleted. I think the problem is in  extracting the .rar files.
In debuglog is shown as follow

04.02.2015 11:12:08 DEBUG     ExtractArchive: Extracted files: [u'/home/svenilee/.pyload/Downloads/MaHe0202/xxx.mkv']
04.02.2015 11:12:08 DEBUG     ExtractArchive: MaHe0202.part2.rar | Skipped
04.02.2015 11:12:08 DEBUG     ExtractArchive: MaHe0202.part3.rar | Skipped
04.02.2015 11:12:08 DEBUG     ExtractArchive: MaHe0202.part4.rar | Skipped
04.02.2015 11:12:08 DEBUG     ExtractArchive: MaHe0202.part5.rar | Skipped
04.02.2015 11:12:08 DEBUG     ExtractArchive: MaHe0202.part6.rar | Skipped

**Comment by immenz** [comment]
Yes, extract part1 of multipart means extraction of all others to. But this is logic done by the extractor programm, in pyload we only can work with the filenames there are... I know it worked before, I will look in older versions how they handled it and try not to brake new stuff^^

**Comment by immenz** [comment]
Found why pyload only delete one file...such a little thing ^^

**Comment by sveni-lee** [comment]
cool... would you share?

**Comment by immenz** [comment]
https://github.com/pyload/pyload/pull/1141

**Comment by immenz** [comment]
I would still recommend not to use the ExtractArchive. When downloading more than one package, the plugin act weird on my PC. 
After several test runs the file.db seems to save hole files, because it's size extends over 200 mb. I think relatet to that is the fact that only last package in a set of different packages downloaded in a row is actually extracted.

So hold on a little wile :/

**Comment by wastez** [comment]
Will there be a working version ever again???

**Comment by sveni-lee** [comment]
thanks for sharing, extracting and deleting works for me... Now I've to get Filebot wirking again but for this I will ask the autor...
Thanks again

**Comment by Killer1337** [comment]
Are your changes merged to the official repo? Cuz I still have an issue with password protected Files. I get the info that no password was found although it is in the passwords.txt and the passwords.txt is correctly linked in the plugin settings.

**Comment by immenz** [comment]
No, still under Pull Requests

**Comment by zlep** [comment]
@immenz 
Thank you. I replaced the ExtractArchive.py and the UnRar.py with your files.
Now I'm getting
ERROR	Error executing hooks: invalid literal for int() with base 10: '\'\\\'\\\\\\\'\\\\\\\\\\\\\\\'"\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\'42\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\'"\\\\\\\\\\\\\\\'\\\\\\\'\\\'\''

overtime it finishes a package and should start extracting. Not sure if this has to do with the changes you made?! But it does not extract.

**Comment by immenz** [comment]
Yeah, that was part of the problem...or you can say bad remains if you used the broken version of the plugin earlier.

for a quick solution: Are you familiar with sqllite and know how to edit a *.db file? The entry starting with 'ExtractArchive' in the Table 'storage' in the files.db (in pyload config folder) has to be removed.
You can also stop pyload and delete the files.db, but before new starting pyload, a new user for WebUI has to be created. Note that this will also delete all packages/files currently in your queues.
I'll also could write a quick workaround in the .py file, which later can be deleted again.

**Comment by zlep** [comment]
Awesome. It is finally working again. Thank you very much.
Only thing to test is password in the unrar_passwords.txt.

**Comment by mbosner** [comment]
Any progress with the merge request? Else i would fork and merge some of the pending merge requests - or is that done by someone else already?

**Comment by mbosner** [comment]
This helped a lot:

```
delete from storage;
vacuum;
```

The delete command will take its time.

**Comment by wastez** [comment]
@ immenz 
Your version is working. Thanks for it.
Could you please have a look on the extracting status. (it´s not displayed anymore)

**Comment by Killer1337** [comment]
Thank you @immenz 
Your version works as it should.
Lets hope it get's merged quickly

**Comment by Ghoststage** [comment]
Ihr habt so viel an pyload kaput gemacht warum gebt ihr das gefixte nicht
frei? Mir kommt es so vor als man euch an anbetteln muss damit ihr es wider
sauber hinbekommt

und fügt nicht immer neue hoster hinzu sorgt erst mal dafür das alles
sauber leuft damit meine ich

1.Download von Alle Hoster
2.Extract (Mit STATUS)
3.Nach dem Entpacken Files löschen


Uploaded Zeigt keine Datei Größe an und die namen der dateien werden auch
nicht angezeigt sie sehen so aus http://ul.to/nvr04ah9 sobald der download
startet heissen die dateie so wie es sein soll

2015-02-06 7:11 GMT+01:00 Killer1337 <notifications@github.com>:

**Comment by zlep** [comment]
@Ghoststage Soweit ich weiß ist @immenz kein offizieller Pyload Entwickler. Wir können also dankbar sein, dass wenigstens einer so viel Sachverstand hat und das ExtractArchive.py wieder zum Laufen bekommen hat, nachdem @vuolter es seit Wochen NICHT geschafft hat.
Ich habe jetzt diese Version installiert und den updatemanager ausgeschaltet. Damit dürfte ich hoffentlich wieder für einige Zeit auf der sicheren Seite sein.
Thx @immenz again

**Comment by wastez** [comment]
If now somebody will merge it to source, repair the extracting status and that "Delete after finish" will be stored after restart pyload will be useable again.

**Comment by Ghoststage** [comment]
Oh mein fehler war natürlich nur an die Pyload Entwickler und nicht an die
user die helfen oder eigene scripte coden

2015-02-06 15:31 GMT+01:00 wastez <notifications@github.com>:

**Comment by immenz** [comment]
@wastez 
> If now somebody will merge it to source, repair the extracting status and that "Delete after finish" will be stored after restart pyload will be useable again.

On my system "delete after finished" is stored properly. So maybe there are some issues with another plugin or with your system (access rights etc.). Did you try to edit ExtractArchive Section in your plugin.conf manually?


@Ghoststage & @zlep :  Ich versteh ja dass es ärgerlich ist, dass pyload bzw. einige Plugins plötzlich nicht mehr das tun, was sie vorher taten. Aber ich finde es ist wie mit Wählen gehen: Wer es nicht tut, hat auch kein Recht sich drüber zu beschweren. Jedenfalls nicht so grob. 
Soll heißen: Die "Entwickler" von pyload sind auch nur User, die mit viel Sachverstand und vor allem Zeit ein Programm auf die Beine gestellt haben, dass sie freundlicherweise auch anderen Leuten zur Verfügung stellen, und zwar kostenlos! Dabei haben diese Leute bestimmt auch noch andere Sachen zu tun, als sich 24/7 um das korrekte Verhalten des Programm zu kümmern. Da kommt Github ins Spiel. Hier hat jeder mit den entsprechenden Fähigkeiten die Möglichkeit, Verbesserungen einzubringen oder Fehler zu melden. Bei der Komplexität des Programms ist es den Hauptentwicklern kaum mehr möglich, all ihre Änderungen bzw. Bugfixes auf Herz und Nieren zu testen. Und nichts anderes habe ich gemacht. Ich habe nur ein paar Kleinigkeiten hier und da ausgebessert, die nur durch Testen aufgefallen sind. @vuolter scheint also den Code erst einmal ohne direktes Testen jeder Funktion zu schreiben, was wiederrum ziemlich beeindruckend ist, bedenkt man wie wenig Stellen in dem Code fehlerhaft waren. 
Und nochmal: Das ist ein freies Tool, keiner bekommt Geld dafür und trotzdem wird es ziemlich intensiv weiter entwickelt. Und es muss auch weiter entwickelt werden, weil viele Sachen sich ständig ändern und komplexer werden. Von daher darf man auch nicht davon ausgehen, dass ein einmal sauber laufendes System ewig hält. 
Ich gebe zu, da dass Coden mir selbst Spass macht fällt es mir leichter, ein paar Bugs hier und da zu finden und zu beheben. Aber selbst ich bin nicht in der Lage, alles zu verstehen oder gar zu lösen. Aber wenn man die Leute einfach nur beschimpft sie sollten es doch gefälligst mal richtig machen, würde auch mir irgendwann der Spass daran vergehen.

**tl;dr** selber mitmachen, dann darf man sich auch beschweren.

**Comment by wastez** [comment]
Also tried to edit the plugin.conf but after restart it's set back to default. 
Could it have something to do with the files.db?

In the moment i setted the default Setting in extractarchive.py to true as workarround.

**Comment by Ghoststage** [comment]
Ich gebe dir mit allem recht. Nur wenn man was ändert und es freigegeben
ist sollte es auch gehen wenn nicht sollte man sofort dran arbeiten damit
die User nicht frusten da sollte man sich nicht wundern das man sauer auf
die coder wird

Der Gedanke eines Pyload Entwickler wird so aussehen

Hm
Hmm
mir ist langweilig :c
ach ich code was an pyload rum :D
so fertig gefummelt und freigeben.
Ohh da geht was nicht!
ach egal die anderen user machen das schon und wenn nicht müssen sie warten
bis ich nichts zu tun habe
3 Std Später

Mal sehen wie viele gefrustet sind.
oh doch so wenig
dann ist ja alles ok die wo es nicht geht können ja ein anderes tool nehmen


Am 6. Februar 2015 um 17:40 schrieb immenz <notifications@github.com>:

**Comment by immenz** [comment]
don't think files.db has anything to do with it, because I couldn't find something plugin-setting related in it.
I had a similar problem once with an other plugin. Unfortunately I can't remember what caused it, better say I don't know if the things I did resolved it. First I deleted the section related to that plugin from plugin.conf. Then I rewrote the "__config__" section of the py file line by line. Did you replace the plugin File completely with the file file I changed the last couple days or did you just bring in the changes?

**Comment by stickell** [comment]
I was waiting for @vuolter to merge the pull request because he is the one who is following this issue, but because it's taking long and this is a very important plugin I'll do it.

Guys please write in English, we are not all Germans.

**Comment by zlep** [comment]
> On my system "delete after finished" is stored properly. 
Works for me too.

**Comment by immenz** [comment]
How about status of extracting showing in Web UI? Couldn't test it on my windows machine, because unrar is too fast^^

**Comment by wastez** [comment]
I took your full file. 
Also removed the ExtractArchive from Plugin.conf but that doesn't change anything.

**Comment by wastez** [comment]
Now i tried to set RepairArchive to false an this was also not saved. 
But i'm able to set a destination folder.

**Comment by immenz** [comment]
Can you tell what pyload says what plugins are activated on startup? (In Debug mode right after starting pyload: activated plugins: .....)

**Comment by wastez** [comment]
BypassCaptcha, Checksum, ClickAndLoad, ExternalScripts, ExtractArchive, ImageTyperz, LinkdecrypterCom, PyControlPush, RestartFailed, UnSkipOnFail, UpdateManager, XFileSharingPro

**Comment by immenz** [comment]
Sorry, I'm out of Ideas. In my case, there was the old serienjunkies plugin interference my own serienjunkies plugin I believe, and thats why one overwritten the other ones config.
I guess you also tried to deactivate the ExtractArchive Plugin and restart pyload?

**Comment by immenz** [comment]
Sorry folks, I've just found a new bug ;) (Not extracting Archives in Archives)

@stickell As I am continuing testing ExtractArchive, should I open new Issues for every new found or do we go along in this one?

**Comment by Killer1337** [comment]
the version I pulled from @immenz repo 15 hours ago interestingly works for some archives. For others it doesn't fetch the password from the password file and I don't know why.

I'll switch to debug mode and see if I can find out more.

I just hope people like @vuolter and @immenz are not demotivated by naggers like @Ghoststage. Thank you for dedicating your free time! :)

**Comment by mbosner** [comment]
+1 @immenz @stickell

Great work!

**Comment by stickell** [comment]
> As I am continuing testing ExtractArchive, should I open new Issues for every new found or do we go along in this one?

No need to mess jumping from a bug report to another one. Just continue here.

Merged #1152. Thanks for your work @immenz

**Comment by stickell** [comment]
@Ghoststage and @zlep: if you are such great developers, able to teach us who to develop softwares, why don't you use your incredible skills in computer programming to fix the source code?

Also, if you enjoy trolling please do it in English. Don't you have something to hide, right? Don't tell me you can't troll in other languages than German!

We will no longer tolerate this attitude! You are warned. Collaborators are welcome, trolls are not.

**Comment by zlep** [comment]
> @Ghoststage and @zlep: if you are such great developers, able to teach us who to develop softwares, why don't you use your incredible skills in computer programming to fix the source code?

This attitude is ridiculous. If you don't know the background then shut up.
Just to clarify: I never was rude like you're right now. I reported the bugs for a long time but it got worse and worse. If @immenz wouldn't have sorted out the bugs we still would have a broken plugin.

**Comment by stickell** [comment]
@zlep: maybe you don't understand at all that @vuolter is just a normal user like me or @immenz who had worked a lot making a lot of pull request at his time so he entered in the pyLoad Team. Nobody of the original pyLoad Team is still active! We were just users of pyLoad willing to make this software usable. You have no idea about the situation this software was when I started contributing! Almost any plugin was broken! It was quite unusable and it's thanks to work made by users like me, @vuolter or @immenz that this software started working again.

Now you are making a mess for just **one** plugin broken. You have no idea about the amount of work we have done these years. You are the one who don't know the background and should shut up.

If you can't develop code that's fine for me! You can still help @immenz testing his patches and that's a very useful thing trust me, but moaning like a child this way is totally useless.

This is my last post about this stuff. This is a technical place, any further off-topic message will be deleted.

**Comment by mobo95** [comment]
Repair broken archives should use .rev file
Please if you can, implement this.



07.02.2015 14:42:35	INFO	ExtractArchive: lmldzndlfrzrfntsm0680.part10.rar | Extract to: Downloads/Pirati dei caraibi - La maledizione del forziere fantasma/Pirati dei caraibi - La maledizione del forziere fantasma
07.02.2015 14:46:58	ERROR	ExtractArchive: lmldzndlfrzrfntsm0680.part10.rar | CRC mismatch | Cannot find volume Downloads/Pirati dei caraibi - La maledizione del forziere fantasma/lmldzndlfrzrfntsm0680.part33.rar
?	?	La maledizione del forziere fantasma.2006.1080p.BluRay.x264.sLcR3W.mkv - CRC failed

**Comment by mobo95** [comment]
What does mean this?

97	07.02.2015 14:50:34	INFO	ExtractArchive: catem.part23.rar | Extract to: Downloads/Cate McCall - Il confine della verità/Cate McCall - Il confine della verità
98	07.02.2015 14:50:34	INFO	ExtractArchive: catem.part23.rar | Password protected
99	07.02.2015 14:50:35	ERROR	ExtractArchive: catem.part23.rar | 'ascii' codec can't decode byte 0xc3 in position 60: ordinal not in range(128)

**Comment by wastez** [comment]
Is somebody working on the extraction status in the moment? 
Would be really great if somebody could do this.

**Comment by wastez** [comment]
Most of the time extracting is working fine with passwords.
But on some files no password can be found. (but the password is the same like the working files)

Here the log:
558	08.02.2015 04:33:31	INFO	ExtractArchive: Check package: ahs.x264.S04E09
559	08.02.2015 04:33:31	INFO	ExtractArchive: ahs.x264.S04E09.part3.rar | Extract to: /media/02aa647a-4695-4417-b44d-bafe779f5393/Downloads/Extracted/
560	08.02.2015 04:33:57	ERROR	ExtractArchive: ahs.x264.S04E09.part3.rar | No password found
561	08.02.2015 04:33:57	ERROR	ExtractArchive: ahs.x264.S04E09.part3.rar | Extract failed

**Comment by immenz** [comment]
@wastez 
should be working in the next update (pull request's out)

Tested with your file (ahs...)

**Comment by wastez** [comment]
@immenz 
I took the files from your newest pull request and it worked great. You're the best. :)
What about the storing problem after restart?
I saw that much People have the same issue like me.

**Comment by mobo95** [comment]
when latest changes will be available on update manager?

**Comment by wastez** [comment]
vuolter said today that he will have a look on it.
